pkg/app/piped/cmd/piped/piped.go

```diff
@@ -574,8 +578,9 @@ func (p *piped) createAPIClient(ctx context.Context, address, projectID, pipedID
 
 // loadConfig reads the Piped configuration data from the specified source.
 func (p *piped) loadConfig(ctx context.Context) (*config.PipedSpec, error) {
-	if p.configFile != "" && p.configGCPSecret != "" {
-		return nil, fmt.Errorf("only config-file or config-gcp-secret could be set")
+	// HACK: When the version of cobra is updated to >=v1.8.0, this should be replaced with https://pkg.go.dev/github.com/spf13/cobra#Command.MarkFlagsOneRequired.
+	if err := p.hasTooManyConfigFlags(); err != nil {
```

Comment:
I agree with updating cobra's version 👍 

Btw, the function might not work as you expected 👀 
I think you expected to check whether there is exactly one flag with `MarkFlagsOneRequired`, right?

https://pkg.go.dev/github.com/spf13/cobra#Command.MarkFlagsOneRequired

> MarkFlagsOneRequired marks the given flags with annotations so that Cobra errors if the command is invoked without at least one flag from the given set of flags.

This means that `MarkFlagsOneRequired` returns like below when the given flags are {a, b, c}.
- {}: false
- {a}: true
- {a b}: true
- {a b c}: true
Also, `!MarkFlagsOneRequired()` means all of them are not set 👀

Reply:
Could you open an issue for updating cobra version later?

Reply:
@ffjlabo 
I'm really sorry.
The correct func is [`MarkFlagsMutuallyExclusive`](https://pkg.go.dev/github.com/spf13/cobra#Command.MarkFlagsMutuallyExclusive).

So I fixed [the comment](https://github.com/pipe-cd/pipecd/pull/4785/files#diff-a6436ea98eb0426cf2f50c52d74885ac49f3c87c22f0d2a41c3c1a8975d661ccR581) to below:

> // HACK: When the version of cobra is updated to >=v1.8.0, this should be replaced with https://pkg.go.dev/github.com/spf13/cobra#Command.MarkFlagsMutuallyExclusive.

Reply:
@khanhtc1202 
Yeah, I opened an issue https://github.com/pipe-cd/pipecd/issues/4788 now.

Reply:
Wow, that's good!
I didn't know the method 👀